java/fury-core/src/main/java/org/apache/fury/Fury.java

```diff
@@ -307,23 +309,26 @@ private void write(MemoryBuffer buffer, Object obj) {
     int startOffset = buffer.writerIndex();
     boolean shareMeta = config.isMetaShareEnabled();
     if (shareMeta) {
-      buffer.writeInt32(-1); // preserve 4-byte for nativeObjects start offsets.
+      buffer.writeInt32(
+          Constants.INVALID_OFFSITE); // preserve 4-byte for nativeObjects start offsets.
     }
     // reduce caller stack
-    if (!refResolver.writeRefOrNull(buffer, obj)) {
+    boolean isNull = refResolver.writeRefOrNull(buffer, obj);
+    if (!isNull) {
       ClassInfo classInfo = classResolver.getOrUpdateClassInfo(obj.getClass());
       classResolver.writeClass(buffer, classInfo);
       writeData(buffer, classInfo, obj);
     }
-    if (shareMeta) {
+    if (shareMeta && !isNull) {
       buffer.putInt32(startOffset, buffer.writerIndex());
       classResolver.writeClassDefs(buffer);
     }
   }
 
   private void xserializeInternal(MemoryBuffer buffer, Object obj) {
     int startOffset = buffer.writerIndex();
-    buffer.writeInt32(-1); // preserve 4-byte for nativeObjects start offsets.
+    buffer.writeInt32(
+        Constants.INVALID_OFFSITE); // preserve 4-byte for nativeObjects start offsets.
```

Comment:
This is an internal, could we keep it inside this class

Reply:
Also used in ClassResolver https://github.com/apache/fury/pull/1686/files#diff-849c034dbc915d01d33d09221ec8c78b290ac53a845e3197ab648ec05bea480aR1427-R1430